pkg/cidata/cidata.TEMPLATE.d/boot/08-shell-prompt.sh

```diff
@@ -1,7 +1,9 @@
 #!/bin/sh
 set -eux
 
-if [ "${LIMA_CIDATA_NAME}" = "default" ]; then
+# This script is only intended for the default.yaml image, which is based on Ubuntu LTS
+
+if [ "${LIMA_CIDATA_NAME}" = "default" ] && grep -q color_prompt /etc/skel/.bashrc; then
```

Comment:
Why is this grepping `/etc/skel/.bashrc` and not `/home/${LIMA_CIDATA_USER}.linux/.bashrc`, the file we are actually going to patch?

My concerns about using `patch` at all from https://github.com/lima-vm/lima/pull/486#issuecomment-996532216 remain.

But going further, if this should only ever be applied to the "default" Ubuntu instance, why isn't this a provisioning script in `default.yaml` instead, making it much easier for the end user to tweak?

Reply:
It was relying on the knowledge of it being the template file.

Reply:
I was reluctant to add any scripts to the default ("empty" ?) template

The original goal was just to tweak the prompt when running `lima`:

* https://github.com/lima-vm/lima/issues/427

There are probably other ways to accomplish the same thing...

Reply:
I know this ship has probably sailed, but injecting the following snippet at the end of `/etc/bashrc` would have probably been more than enough to solve the "make the prompt distinguishable" problem:

```sh
[ "$PS1" ] && PS1="[lima]$PS1"
```

Fiddling with colors shouldn't be the responsibility of Lima IMO. It's just impossible to support all OS out there, let alone all their versions, with any kind of PS1 patching approach.

Reply:
As mentioned in other comments, there was never any intentation on patching any other system than default.yaml

Reply:
Note that the colors are _already_ being fiddled with, the idea was just to make it specific to ubuntu-lima distribution ?

But the since the whole idea with "lime" is gone anyway, and we don't have a new one, I'm fully prepared to just drop it...

* https://github.com/lima-vm/lima/issues/268#issuecomment-1069351680

* https://github.com/lima-vm/lima/pull/486

* `#BFFF00` "lime" (154 GreenYellow)

* `#00bf8c` "cyan" (36 DarkCyan)

There was also support for `NO_COLOR`, if you just wanted to get away...

https://no-color.org/

But there's two "lima":

```
$ TERM=xterm lima
anders@lima-default:/home/anders$ [ "$PS1" ] && PS1="[lima]$PS1"
[lima]anders@lima-default:/home/anders$ 
```

Reply:
I don't have a strong opinion on what the prompt should look like. In fact, as a Lima user my preference would be to leave the guest OS' shell config alone entirely for two reasons:

- Probably 90% of people have a customized prompt of some sort on their host, they will notice a change while opening a `lima` terminal. (Good point regarding the Lima hostname in the prompt.) 
- Lima's provisioning scripts are powerful enough for users to decide whether or not they want to apply their own customizations. (Mount/symlink dotfiles, patch configs, ...)

Regarding this PR, it solves the problem with VMs consistently reporting a failed provisioning (via cloud-final unit) so I think it's a useful, low-hanging workaround. Thanks for addressing it!

Reply:
Yeah, that was probably a bug (or lack of paranoia)

Reply:
@jandubois LGTY?